backend/test_app.py

```diff
@@ -0,0 +1,271 @@
+import os
+import unittest
+import json
+from flask_sqlalchemy import SQLAlchemy
+
+from app import create_app
+from models import setup_db, Exercise, User
+
+
+class ExerciseTrackerTestCase(unittest.TestCase):
+    # This class represents the Exercise Tracker test case
+
+    def setUp(self):
+        # Define test variables and initialize app.
+        self.app = create_app()
+        self.client = self.app.test_client
+        self.database_name = "exercise-tracker_test"
+        setup_db(self.app)
+
+        self.new_user = {
+          "username": "manuelar"
+        }
+
+        self.new_exercise = {
+          "userId": "91298d0b-66c9-493b-8c1d-a79bcf7838d7",
+          "exercise_date": "Fri, 15 May 2020 00:00:00 GMT",
+          "description": "My exercise",
+          "duration": "12"
+        }
+
+        # binds the app to the current context
+        with self.app.app_context():
```

Comment:
We should make a DB mock (SQLAlchemy mock in this case) instead of loading the db connection. When doing unit-tests we should test only what is under our control. A db is classified as an external dependency.